libp2p/Host.h

```diff
@@ -178,20 +191,28 @@ class Host: public Worker
 
 	void registerPeer(std::shared_ptr<Session> _s, CapDescs const& _caps);
 
-	std::shared_ptr<Node> node(NodeId _id) const { if (m_nodes.count(_id)) return m_nodes.at(_id); return std::shared_ptr<Node>(); }
+//	std::shared_ptr<PeerInfo> node(NodeId _id) const { if (m_nodes.count(_id)) return m_nodes.at(_id); return std::shared_ptr<PeerInfo>(); }
+
+protected:
+	void onNodeTableEvent(NodeId _n, NodeTableEventType _e);
 
 private:
 	/// Populate m_peerAddresses with available public addresses.
 	void determinePublic(std::string const& _publicAddress, bool _upnp);
 	
+	void connect(std::shared_ptr<PeerInfo> const& _n);
+	
+	/// Ping the peers to update the latency information and disconnect peers which have timed out.
+	void keepAlivePeers();
+	
 	/// Called only from startedWorking().
 	void runAcceptor();
 	
+	/// Handler for verifying handshake siganture before creating session. _nodeId is passed for outbound connections.
```

Comment:
If `_socket` stays a plain pointer, please put ownership information into the documentation.